expertsystem/ui/__init__.py

```diff
@@ -74,9 +101,9 @@ def __init__(  # pylint: disable=too-many-arguments
                 f"Formalism type {formalism_type} not implemented."
                 f" Use {allowed_formalism_types} instead."
             )
-        self.__formalism_type = formalism_type
-        self.number_of_threads = number_of_threads
-        self.propagation_mode = propagation_mode
+        self.__formalism_type = str(formalism_type)
+        self.number_of_threads = int(number_of_threads)
+        self.propagation_mode = str(propagation_mode)
```

Comment:
Should also the types of those instance attributes be defined?

Reply:
`mypy` deduces them from the cast